src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1200,114 +1201,111 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
 
         // BROKER flow intercepts here
         // cache and refresh call happens through the authenticator service
-        if (mBrokerProxy.canSwitchToBroker()
-                && mBrokerProxy.verifyUser(request.getLoginHint(),
-                        request.getUserId())) {
-            Logger.v(TAG, "It switched to broker for context: " + mContext.getPackageName());
-            AuthenticationResult result = null;
-            request.setVersion(getVersionName());
-            request.setBrokerAccountName(request.getLoginHint());
-            
-            //check if the redirectUri is valid
-            try
-            {
-                verifyBrokerRedirectUri(request);
-            }
-            catch(AuthenticationException exception)
-            {
-                Logger.v(TAG + methodName, "Did not pass the verification of the broker redirect URI");
-                callbackHandle.onError(exception);
-                return result;
-            }
-            
-            // Don't send background request, if prompt flag is always or
-            // refresh_session
-            if (!promptUser(request.getPrompt())
-                    && (!StringExtensions.IsNullOrBlank(request.getBrokerAccountName()) || !StringExtensions
-                            .IsNullOrBlank(request.getUserId()))) {
-                try {
-                    Logger.v(TAG, "User is specified for background token request");
-                    result = mBrokerProxy.getAuthTokenInBackground(request);
-                } catch (AuthenticationException ex) {
-                    // pass back to caller for known exceptions such as failure
-                    // to encrypt
-                    if (callbackHandle.callback != null) {
-                        callbackHandle.onError(ex);
-                        return null;
-                    } else {
-                        throw ex;
+        try {
```

Comment:
At some point i think we should refactor this logic. Having a big try catch here is not an ideal way.